why-fly.qmd

```diff
@@ -0,0 +1,82 @@
+| Docker-compose                                               | Yes             | No (but has render.yml)              | Yes (with a normal node) | No                     | No                          |
+| Logging                                                      | Very good logs  | Simple                               | Simple                   | Simple                 | Yes                         |
+| Alert                                                        | Yes             | No                                   | ?                        | No                     | ?                           |
+| Similar to production environment (Aarhus University server) | Likely possible | Maybe if docker on university server | Likely                   | No                     | Yes if docker               |
+
+## Decision Outcome
+
+We have decided on using fly.io for our demo and testing needs. Fly is very cheap and the cost is not based on the number of users - only the actual usage (around 1$). It integrates with GitHub authentication and the deployment is easy to setup with a Github action. This part is less user-friendly than **render** and **vercel**, where the deployment happens automatically. However, we would eventually adjust these "magic deployment" anyway to do test and etc before deploying.
+
+fly.io uses docker images, so there is no "native" support for a Django application, but flyctl automatically creates a Dockerfile which makes it easy anyway. 
+
+fly.io automatically and relatively quickly scales up an app when needed, which makes it very cheap. Furthermore, fly.io owns all the servers, which makes it cheap compared to other providers which are usually relying on AWS or Azure.
+
+fly.io has a customisable metrics dashboard (every app has access to Prometheus and Grafana)
+
+The other cloud providers, where not chosen for these reasons:
+
+- Render.com was our initial choice. It was cheap and easy to setup, but the cost is per user plus the usage (around 100$ per month vs 1$ for fly.io). Render.com is very user friendly with magic deployments, but we would need to adjust these anyway for test and etc. 
+- Azure is able to do everything we require, but it is too complicated for a demo environment.
+- Vercel is easy to use, but lacks customisability. For instance, it seemed not to work so well with Poetry and it is unable to work with Docker.
+- Digital Ocean could be a fair choice, but I was unable to make it work - so in that sense it fails the "ease of use"-criteria.
+
+### Consequences
+
+- PostgreSQL is a natural part of fly.io, but it is not a **Managed PostgreSQL** like the other cloud providers. A PostgreSQL is "just an app" in fly.io, so it might require a bit more maintenance. But even if everything is lost (extreme case), then it is okay because fly.io is only used as a demo environment
```

Comment:
```suggestion
- PostgreSQL is a natural part of Fly.io, but it is not a **Managed PostgreSQL** like the other cloud providers. A PostgreSQL is "just an app" in Fly.io, so it might require a bit more maintenance. However, even in the unlikely scenario of complete data loss, there is no significant impact since Fly.io serves solely as a demonstration environment.
```